src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -426,8 +426,34 @@ typedef struct MPIDI_OFI_win_request {
     struct fi_context context[MPIDI_OFI_CONTEXT_STRUCTS];       /* fixed field, do not move */
     int event_id;               /* fixed field, do not move */
     struct MPIDI_OFI_win_request *next;
+    struct MPIDI_OFI_win_request *prev;
     int target_rank;
     int rma_type;
+    union {
+        struct {
+            /* origin */
+            const void *origin_addr;
+            int origin_count;
+            MPI_Datatype origin_datatype;
+            size_t origin_bytes;
+            MPI_Aint origin_pack_offset;
+            void *pack_buffer;
+            MPI_Aint pack_size;
+            /* target */
```

Comment:
Can these be separated into sub structures, one for the origin and one for the target?  Just a preference thing, not important if you prefer this format.

Reply:
Can do.

Reply:
Done.